app/main/views/services.py

```diff
@@ -119,72 +123,96 @@ def edit(service_id, section_id):
-)
+@main.route('/services/<service_id>/updates', methods=['GET'])
 @login_required
 @role_required('admin', 'admin-ccs-category')
-def compare(old_archived_service_id, new_archived_service_id):
-
-    def validate_archived_services(old_archived_service, new_archived_service):
-
-        if old_archived_service.get('id', -1) \
-                != new_archived_service.get('id', -2):
-            return False
-
-        old_updated_at = datetime.strptime(
-            old_archived_service.get('updatedAt'), DATETIME_FORMAT)
-
-        new_updated_at = datetime.strptime(
-            new_archived_service.get('updatedAt'), DATETIME_FORMAT)
-
-        if old_updated_at >= new_updated_at:
-            return False
-
-        return True
-
-    try:
-        service_data_revision_1 = data_api_client.get_archived_service(
-            old_archived_service_id)['services']
-
-        service_data_revision_2 = data_api_client.get_archived_service(
-            new_archived_service_id)['services']
-
-        # ids exist, ids match, dates are chronological
-        if not validate_archived_services(
-                service_data_revision_1, service_data_revision_2):
-            raise ValueError
-
-        service_data = data_api_client.get_service(
-            service_data_revision_1['id'])['services']
+def service_updates(service_id):
+    service_response = data_api_client.get_service(service_id)
+    if service_response is None:
+        abort(404)
+    service = service_response['services']
 
-    except (HTTPError, KeyError, ValueError):
-        return abort(404)
+    supplier = data_api_client.get_supplier(service["supplierId"])["suppliers"]
 
-    content = content_loader.get_manifest('g-cloud-6', 'edit_service_as_admin').filter(service_data)
+    common_request_kwargs = {
+        "object_id": service_id,
+        "object_type": "services",
+        "audit_type": AuditTypes.update_service,
+        "acknowledged": "false",
+    }
 
-    # It's possible to have an empty array if none of the lines were changed.
-    # TODO This possibility isn't actually handled.
-    service_diffs = get_diffs_from_service_data(
-        sections=content.sections,
-        revision_1=service_data_revision_1,
-        revision_2=service_data_revision_2,
-        include_unchanged_lines_in_output=False
+    all_update_events = latest_update_events = oldest_update_events = None
+    latest_update_events_response = data_api_client.find_audit_events(
+        latest_first="true",
+        **common_request_kwargs
     )
+    latest_update_events = latest_update_events_response["auditEvents"]
+
+    if latest_update_events_response["links"].get("next"):
+        # we haven't got all update events for this object. fetch the oldest
+        oldest_update_events_response = data_api_client.find_audit_events(
+            latest_first="false",
+            **common_request_kwargs
+        )
+        oldest_update_events = oldest_update_events_response["auditEvents"][::-1]
```

Comment:
I don't know if it's sensible to reverse the order of these events. I would expect `latest_update_events` to have the most recent event at `0`, which it does, and I would expect `oldest_update_events` to have the oldest event at `0`, which it no longer does. It has the most recent event at `0` after this mutation. Unless there's a good reason for doing this, it's probably better to leave it in the order it's received from the API or it's going to be easy to be confused about the order.